src/KnownProperties.jl

```diff
@@ -0,0 +1,82 @@
+########################################################################
+# This is to introduce a common language to ask whether or not a given
+# property (which is potentially hard to compute) is known for some
+# object `x`. 
+########################################################################
+
+@doc raw"""
+    is_known(x::Any, f::Function)
+
+Return whether a property of an object `x` called for by `f(x)` is known. 
```

Comment:
"is known" IMHO does not convey any clear meaning. The bit we care about is "accessing this can be done in constant time", so let's state that.


I also don't think that the meaning of "property" is universally clear here; in particular it is not the usual Julia meaning (as in `getproperty` and `setproperty`).

How about
```suggestion
Given a unary function `f` and a value `x`, return whether `f(x)` is "known"
in the sense that evaluating `f(x)` is fast and takes constant time.

For example this might return `true` if `f(x)` was computed before and has
been cached.
```